BlueBridgeCore/src/main/java/de/mark225/bluebridge/core/bluemap/BlueMapIntegration.java

```diff
@@ -3,102 +3,86 @@
 import com.flowpowered.math.vector.Vector2d;
 import com.flowpowered.math.vector.Vector3d;
 import de.bluecolored.bluemap.api.BlueMapAPI;
-import de.bluecolored.bluemap.api.BlueMapAPIListener;
 import de.bluecolored.bluemap.api.BlueMapMap;
-import de.bluecolored.bluemap.api.BlueMapWorld;
-import de.bluecolored.bluemap.api.marker.*;
+import de.bluecolored.bluemap.api.markers.DistanceRangedMarker;
+import de.bluecolored.bluemap.api.markers.ExtrudeMarker;
+import de.bluecolored.bluemap.api.markers.MarkerSet;
+import de.bluecolored.bluemap.api.markers.ShapeMarker;
+import de.bluecolored.bluemap.api.math.Shape;
 import de.mark225.bluebridge.core.BlueBridgeCore;
 import de.mark225.bluebridge.core.addon.AddonRegistry;
 import de.mark225.bluebridge.core.addon.BlueBridgeAddon;
-import de.mark225.bluebridge.core.config.BlueBridgeConfig;
 import de.mark225.bluebridge.core.region.RegionSnapshot;
 import de.mark225.bluebridge.core.update.UpdateTask;
 import org.apache.commons.lang.StringEscapeUtils;
 import org.bukkit.Bukkit;
-import org.bukkit.util.StringUtil;
 
-import java.io.IOException;
 import java.util.*;
-import java.util.logging.Level;
 import java.util.stream.Collectors;
 
 public class BlueMapIntegration {
 
     private BlueMapAPI blueMapAPI = null;
 
-    public void onEnable(BlueMapAPI blueMapAPI) {
+    public void onEnable(BlueMapAPI blueMapAPI){
         this.blueMapAPI = blueMapAPI;
         Bukkit.getScheduler().runTask(BlueBridgeCore.getInstance(), () ->{
             BlueBridgeCore.getInstance().updateConfig();
             BlueBridgeCore.getInstance().reloadAddons();
             resetMarkers();
             UpdateTask.worlds.clear();
             UpdateTask.resetLastSnapshots();
-            UpdateTask.worlds.addAll(blueMapAPI.getWorlds().stream().map(BlueMapWorld::getUuid).collect(Collectors.toList()));
+            UpdateTask.worlds.addAll(blueMapAPI.getWorlds().stream().map(blueMapWorld -> UUID.fromString(blueMapWorld.getId())).collect(Collectors.toList()));
```

Comment:
You can't rely on `BlueMapWorld#getId()` being a UUID .. it is in most cases, but it's very well possible to be something else theoretically

Reply:
What else on Bukkit :thinking:

Reply:
It is better to map it the other way around. Instead of trying to get the Bukkit World UUID from a BlueMapWorld, try to get the BlueMap World for each loaded Bukkit World and collect the ones which are present. Then save them in a HashMap mapping UUIDs to BlueMapWorld Objects if you need them later.  
If you call `BlueMapAPI.getWorld(uuid)` BlueMap will reliably return the BlueMapWorld for that Bukkit World UUID. But the other way around it might change at some point since the API does not guarantee you to return a UUID String on `BlueMapWorld.getId()`.  
So basically both approaches work fine, but this might not be as future-proof as the alternative.

Reply:
Thanks to both of you!